File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        ht = {}

        for num in nums:
            ht[num] = 1 + ht.get(num, 0)
        
        heap = []
        for num in ht.keys():
            heapq.heappush(heap, [-ht[num], num])
        
        #res = []
        # for i in range(k):
        #     res.append(heapq.heappop(heap)[1])
        res = [heapq.heappop(heap)[1] for _ in range(k)]
        return res
            
        

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> hset;

        for (auto num : nums){
            if( hset.count(num))
                return !false;
            hset.insert(num);
        }
        return !true;
    }
};